src/java.base/share/classes/javax/crypto/spec/HKDFParameterSpec.java

```diff
@@ -0,0 +1,503 @@
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javax.crypto.spec;
+
+import jdk.internal.javac.PreviewFeature;
+
+import javax.crypto.SecretKey;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Parameters for the combined Extract, Expand, or Extract-then-Expand
+ * operations of the HMAC-based Key Derivation Function (HKDF). The HKDF
+ * function is defined in <a href="http://tools.ietf.org/html/rfc5869">RFC
+ * 5869</a>.
+ * <p>
+ * In the Extract and Extract-then-Expand cases, the {@code addIKM} and
+ * {@code addSalt} methods may be called repeatedly (and chained). This provides
+ * for use-cases where a {@code SecretKey} may reside on an HSM and not be
+ * exportable. The caller may wish to provide a label (or other components) of
+ * the IKM without having access to the portion stored on the HSM. The same
+ * feature is available for salts.
+ * <p>
+ * The above feature is particularly useful for "labeled" HKDF Extract used in
+ * TLS 1.3 and HPKE, where the IKM consists of concatenated components, which
+ * may include both byte arrays and (possibly non-extractable) secret keys.
+ * <p>
+ * Examples:
+ * {@snippet lang = java:
+ *
+ * AlgorithmParameterSpec kdfParameterSpec =
+ *             HKDFParameterSpec.ofExtract()
+ *                              .addIKM(ikmPart1)
+ *                              .addIKM(ikmPart2)
+ *                              .addSalt(salt).extractOnly();
+ *
+ *
+ *}
+ * {@snippet lang = java:
+ *
+ * AlgorithmParameterSpec kdfParameterSpec = HKDFParameterSpec.expandOnly(prk,
+ * info, 32);
+ *
+ *}
+ * {@snippet lang = java:
+ *
+ * AlgorithmParameterSpec kdfParameterSpec =
+ *             HKDFParameterSpec.ofExtract()
+ *                              .addIKM(ikm)
+ *                              .addSalt(salt).thenExpand(info, 42);
+ *
+ *}
+ *
+ * @since 24
+ */
+@PreviewFeature(feature = PreviewFeature.Feature.KEY_DERIVATION)
+public interface HKDFParameterSpec extends AlgorithmParameterSpec {
```

Comment:
Does this class need to extend `AlgorithmParameterSpec`? It's more like a factory.

Reply:
OK.